From: Mike Hommey <mh+mozilla@glandium.org>
Date: Fri, 12 Jul 2013 09:14:20 +0900
Subject: Bug 892366 - Support R_ARM_JUMP24 relocations in elfhack. r=nfroyd

---
 build/unix/elfhack/elfhack.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/build/unix/elfhack/elfhack.cpp b/build/unix/elfhack/elfhack.cpp
index 712a749..55268d7 100644
--- a/build/unix/elfhack/elfhack.cpp
+++ b/build/unix/elfhack/elfhack.cpp
@@ -16,6 +16,9 @@
 #ifndef R_ARM_V4BX
 #define R_ARM_V4BX 0x28
 #endif
+#ifndef R_ARM_JUMP24
+#define R_ARM_JUMP24 0x1d
+#endif
 #ifndef R_ARM_THM_JUMP24
 #define R_ARM_THM_JUMP24 0x1e
 #endif
@@ -326,6 +329,7 @@ private:
             case REL(ARM, REL32):
                 apply_relocation<pc32_relocation>(the_code, buf, &*r, addr);
                 break;
+            case REL(ARM, JUMP24):
             case REL(ARM, PLT32):
                 apply_relocation<arm_plt32_relocation>(the_code, buf, &*r, addr);
                 break;
