From: Mike Hommey <mh@glandium.org>
Date: Sat, 2 Aug 2014 09:11:01 +0900
Subject: Bug 1025689 - Disable some libopus ARM features on < ARMv6

---
 media/libopus/moz.build | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/media/libopus/moz.build b/media/libopus/moz.build
index 2761e9e..2607a19 100644
--- a/media/libopus/moz.build
+++ b/media/libopus/moz.build
@@ -28,10 +28,11 @@ if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_AS']:
     DEFINES['OPUS_ARM_ASM'] = True
     DEFINES['OPUS_ARM_EXTERNAL_ASM'] = True
     DEFINES['OPUS_ARM_INLINE_ASM'] = True
-    DEFINES['OPUS_ARM_INLINE_EDSP'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_EDSP'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_MEDIA'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_NEON'] = True
+    if int(CONFIG['ARM_ARCH']) >= 6:
+        DEFINES['OPUS_ARM_INLINE_EDSP'] = True
+        DEFINES['OPUS_ARM_MAY_HAVE_EDSP'] = True
+        DEFINES['OPUS_ARM_MAY_HAVE_MEDIA'] = True
+        DEFINES['OPUS_ARM_MAY_HAVE_NEON'] = True
 
 if CONFIG['MOZ_DEBUG']:
     DEFINES['ENABLE_ASSERTIONS'] = True
