From: Mike Hommey <mh+mozilla@glandium.org>
Date: Mon, 29 Jul 2013 09:15:16 +0900
Subject: Bug 898916 - Properly align statically allocated classinfo objects

---
 xpcom/glue/nsIClassInfoImpl.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/xpcom/glue/nsIClassInfoImpl.h b/xpcom/glue/nsIClassInfoImpl.h
index 87c8230..85b556b 100644
--- a/xpcom/glue/nsIClassInfoImpl.h
+++ b/xpcom/glue/nsIClassInfoImpl.h
@@ -7,6 +7,7 @@
 
 #include "nsIClassInfo.h"
 #include "nsISupportsImpl.h"
+#include "mozilla/Util.h"
 
 #include NEW_H
 
@@ -119,13 +120,13 @@ private:
     _flags | nsIClassInfo::SINGLETON_CLASSINFO,                         \
     _cid,                                                               \
   };                                                                    \
-  static char k##_class##ClassInfoDataPlace[sizeof(GenericClassInfo)];  \
+  mozilla::AlignedStorage2<GenericClassInfo> k##_class##ClassInfoDataPlace;   \
   nsIClassInfo* NS_CLASSINFO_NAME(_class) = NULL;
 
 #define NS_IMPL_QUERY_CLASSINFO(_class)                                       \
   if ( aIID.Equals(NS_GET_IID(nsIClassInfo)) ) {                              \
     if (!NS_CLASSINFO_NAME(_class))                                           \
-      NS_CLASSINFO_NAME(_class) = new (k##_class##ClassInfoDataPlace)         \
+      NS_CLASSINFO_NAME(_class) = new (k##_class##ClassInfoDataPlace.addr())  \
         GenericClassInfo(&k##_class##ClassInfoData);                          \
     foundInterface = NS_CLASSINFO_NAME(_class);                               \
   } else
