From: Mike Hommey <mh@glandium.org>
Date: Mon, 27 Dec 2010 10:44:28 +0100
Subject: Add a 2 minutes timeout on xpcshell tests

---
 testing/xpcshell/runxpcshelltests.py | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/testing/xpcshell/runxpcshelltests.py b/testing/xpcshell/runxpcshelltests.py
index 000968d..6a1a9f0 100644
--- a/testing/xpcshell/runxpcshelltests.py
+++ b/testing/xpcshell/runxpcshelltests.py
@@ -10,6 +10,7 @@ import math
 import os
 import os.path
 import random
+import select
 import shutil
 import signal
 import socket
@@ -594,7 +595,21 @@ class XPCShellTestThread(Thread):
             if self.interactive:
                 self.log.info("TEST-INFO | %s | Process ID: %d" % (name, proc.pid))
 
-            stdout, stderr = self.communicate(proc)
+            if self.pStdout == PIPE:
+                stdout = ""
+                while True:
+                    (r, w, e) = select.select([proc.stdout], [], [], 120)
+                    if len(r) == 0:
+                        stdout += "TEST-UNEXPECTED-FAIL | %s | application timed out after 120 seconds with no output" % (test)
+                        proc.kill()
+                        break
+                    line = proc.stdout.read(1)
+                    if line == "":
+                        break
+                    stdout += line
+                proc.wait()
+            else:
+                stdout, stderr = self.communicate(proc)
 
             if self.interactive:
                 # Not sure what else to do here...
