From: Mike Hommey <glandium@debian.org>
Date: Fri, 20 Aug 2010 18:43:18 +0200
Subject: [PATCH] Add a 2 minutes timeout on xpcshell tests

---
 testing/xpcshell/head.js |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/testing/xpcshell/head.js b/testing/xpcshell/head.js
index deb3618..4ef35c0 100644
--- a/testing/xpcshell/head.js
+++ b/testing/xpcshell/head.js
@@ -79,9 +79,15 @@ function _do_main() {
   var thr = Components.classes["@mozilla.org/thread-manager;1"]
                       .getService().currentThread;
 
+  var timer = Components.classes["@mozilla.org/timer;1"]
+                        .createInstance(Components.interfaces.nsITimer);
+  timer.initWithCallback(function () { _passed = false; dump("TEST-UNEXPECTED-FAIL | test is taking more than 2 minutes\n"); _do_quit() }, 120000, timer.TYPE_ONE_SHOT);
+
   while (!_quit)
     thr.processNextEvent(true);
 
+  timer.cancel();
+
   while (thr.hasPendingEvents())
     thr.processNextEvent(true);
 }
