From: Mike Hommey <glandium@debian.org>
Date: Wed, 12 Mar 2008 23:20:52 +0100
Subject: Don't build example component

---
 xulrunner/moz.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xulrunner/moz.build b/xulrunner/moz.build
index 10dc040..bbe386d 100644
--- a/xulrunner/moz.build
+++ b/xulrunner/moz.build
@@ -9,7 +9,6 @@ CONFIGURE_SUBST_FILES += ['installer/Makefile']
 PARALLEL_DIRS += [
     'app',
     'stub',
-    'examples',
 ]
 
 if CONFIG['MAKENSISU']:
