From: Mike Hommey <glandium@debian.org>
Date: Wed, 12 Mar 2008 23:20:52 +0100
Subject: Don't build example component

---
 xulrunner/moz.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xulrunner/moz.build b/xulrunner/moz.build
index aeb1e94..4799e1d 100644
--- a/xulrunner/moz.build
+++ b/xulrunner/moz.build
@@ -10,7 +10,6 @@ PARALLEL_DIRS += [
     'app',
     'setup',
     'stub',
-    'examples',
 ]
 
 if CONFIG['MAKENSISU']:
