From: Matt Woodrow <mwoodrow@mozilla.com>
Date: Mon, 5 May 2014 13:32:22 +1200
Subject: Backout bug 991767 for causing bug 1003707. a=lsblakk

---
 gfx/src/nsDeviceContext.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/gfx/src/nsDeviceContext.cpp b/gfx/src/nsDeviceContext.cpp
index 07620e7..6dcc97c 100644
--- a/gfx/src/nsDeviceContext.cpp
+++ b/gfx/src/nsDeviceContext.cpp
@@ -405,11 +405,7 @@ nsDeviceContext::CreateRenderingContext(nsRenderingContext *&aContext)
 #endif
     nsRefPtr<nsRenderingContext> pContext = new nsRenderingContext();
 
-    RefPtr<gfx::DrawTarget> dt =
-      gfxPlatform::GetPlatform()->CreateDrawTargetForSurface(printingSurface,
-                                                             gfx::IntSize(mWidth, mHeight));
-
-    pContext->Init(this, dt);
+    pContext->Init(this, printingSurface);
     pContext->Scale(mPrintingScale, mPrintingScale);
     aContext = pContext;
     NS_ADDREF(aContext);
