From: Mike Hommey <mh@glandium.org>
Date: Tue, 26 May 2015 19:06:31 +0900
Subject: Bug 1168231 - Fixup to keep file type

---
 python/mozbuild/mozpack/files.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/python/mozbuild/mozpack/files.py b/python/mozbuild/mozpack/files.py
index 1bd0723..3cddcc5 100644
--- a/python/mozbuild/mozpack/files.py
+++ b/python/mozbuild/mozpack/files.py
@@ -214,12 +214,14 @@ class File(BaseFile):
         assert self.path is not None
         mode = os.stat(self.path).st_mode
         # Normalize file mode:
-        # - take the user bits only
+        # - take the user bits only (leaving away sticky bit, setuid, setgid)
         # - expand them to group and other
         # - apply a standard umask
-        mode = mode & 0700;
+        # - keep file type (e.g. S_IFREG)
+        file_type = stat.S_IFMT(mode)
+        mode = mode & stat.S_IRWXU;
         mode = mode | (mode >> 3) | (mode >> 6)
-        return mode & 0755;
+        return file_type | (mode & 0755);
 
 class ExecutableFile(File):
     '''
