From: Mike Hommey <mh@glandium.org>
Date: Mon, 31 Aug 2015 08:01:39 +0900
Subject: Fixup for bug 1178266

---
 build/autoconf/toolchain.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/autoconf/toolchain.m4 b/build/autoconf/toolchain.m4
index 85a5847..90d8528 100644
--- a/build/autoconf/toolchain.m4
+++ b/build/autoconf/toolchain.m4
@@ -232,15 +232,15 @@ if test "$GNU_CXX"; then
              #include <atomic>],
             [ std::atomic<uint64_t> foo; foo = 1; ],
             ac_cv_needs_atomic=no,
-            _SAVE_LDFLAGS="$LDFLAGS"
-            LDFLAGS="$LDFLAGS -latomic"
+            _SAVE_LIBS="$LIBS"
+            LIBS="$LIBS -latomic"
             AC_TRY_LINK(
                 [#include <cstdint>
                  #include <atomic>],
                 [ std::atomic<uint64_t> foo; foo = 1; ],
                 ac_cv_needs_atomic=yes,
                 ac_cv_needs_atomic="do not know; assuming no")
-            LDFLAGS="$_SAVE_LDFLAGS"
+            LIBS="$_SAVE_LIBS"
         )
     )
     if test "$ac_cv_needs_atomic" = yes; then
