From: Mike Hommey <mh@glandium.org>
Date: Fri, 28 Aug 2015 17:46:29 +0900
Subject: Followup to bug 1194905: add -mfloat-abi=softfp when toolchain abi is
 soft

---
 configure.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.in b/configure.in
index ef275d1..fab7ac0 100644
--- a/configure.in
+++ b/configure.in
@@ -5527,6 +5527,12 @@ if test -n "$MOZ_VPX" -a -z "$MOZ_NATIVE_LIBVPX"; then
         VPX_AS_CONVERSION='$(PERL) $(topsrcdir)/media/libvpx/build/make/ads2gas.pl'
         VPX_ASM_SUFFIX="$ASM_SUFFIX"
         VPX_ARM_ASM=1
+        AC_TRY_COMPILE([],
+          [#ifndef __SOFTFP__
+           #error "float-abi is either softfp or hardfp"
+           #endif],
+           VPX_ASFLAGS="$VPX_ASFLAGS -mfloat-abi=softfp"
+           )
       fi
     ;;
     *:x86)
